# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/layers/wr/WebRenderUserData.h
# Commit: 96da9d241051
# Full Hash: 96da9d241051d222bdd693bdb519d058a013b59c
# Author: Doug Thayer <dothayer@mozilla.com>
# Date: 2019-03-23 09:46:24
# Regressor Bug: 1441308
# File Overlap Count: 2
# Description:
#   Bug 1441308 - Core renderroot splitting changes r=kats,sotaro
#   
#   This is a large patch that contains all of the core changes for
#   renderroot splitting.
#   
# ==============================================================================

diff -r 8cefa694f811 -r 96da9d241051 gfx/layers/wr/WebRenderUserData.h
--- a/gfx/layers/wr/WebRenderUserData.h	Fri Mar 22 18:28:31 2019 +0000
+++ b/gfx/layers/wr/WebRenderUserData.h	Fri Mar 22 18:28:42 2019 +0000
@@ -12,6 +12,7 @@
 #include "mozilla/layers/StackingContextHelper.h"
 #include "mozilla/webrender/WebRenderAPI.h"
 #include "mozilla/layers/AnimationInfo.h"
+#include "mozilla/layers/RenderRootBoundary.h"
 #include "nsIFrame.h"
 #include "ImageTypes.h"
 
@@ -79,6 +80,7 @@
     eCanvas,
     eGroup,
     eMask,
+    eRenderRoot,
   };
 
   virtual UserDataType GetType() = 0;
@@ -245,6 +247,21 @@
   UniquePtr<WebRenderCanvasRendererAsync> mCanvasRenderer;
 };
 
+class WebRenderRenderRootData : public WebRenderUserData {
+ public:
+  WebRenderRenderRootData(RenderRootStateManager* aManager,
+                          nsDisplayItem* aItem);
+  virtual ~WebRenderRenderRootData();
+
+  virtual UserDataType GetType() override { return UserDataType::eRenderRoot; }
+  static UserDataType Type() { return UserDataType::eRenderRoot; }
+
+  RenderRootBoundary& EnsureHasBoundary(wr::RenderRoot aChildType);
+
+ protected:
+  Maybe<RenderRootBoundary> mBoundary;
+};
+
 extern void DestroyWebRenderUserDataTable(WebRenderUserDataTable* aTable);
 
 struct WebRenderUserDataProperty {